### DISCUSSION TITLE
Inheriting Fl_Check_Browser from Fl_Browser (not Fl_Browser_)

### DISCUSSION THREAD

**Discussion by MoAlyousef** [fltk/fltk#1009] [Development]
Hello

I was wondering if it’s possible for Fl_Check_Browser to inherit directly from Fl_Browser instead of Fl_Browser_. 

I have tried to apply the change and the build works. I’m wondering if there are any gotchas I’m probably missing. 

The reason is that it would simplify the inheritance hierarchy for bindings such as the Rust bindings for example. Whereas for example all other browsers inherit from Fl_Browser except for Fl_Check_Browser. 

Thank you

**Comment by Albrecht-S** [comment]
Your proposal looks sensible, but ...
> I’m wondering if there are any gotchas I’m probably missing.

Changing the inheritance of classes would potentially break user programs that use or - more likely - extend FLTK classes, i.e. derive their own subclasses. I'm guessing that the proposed modification would be fine if ...
- Fl_Browser does not override methods of Fl_Browser_ in incompatible ways
- Fl_Browser does not add "unexpected" methods from the POV of a user class inheriting Fl_Check_Browser
- there are no other (yet unknown) issues related to, for instance, virtual methods.

That said, I'm not able to check the details right now, hence I didn't look at the code yet. Therefore it may be possible or not, I'm not sure.

PS(OT): :+1: Thanks for using our new GitHub Discussions option (you're the first FLTK user posting here).

**Reply by MoAlyousef** [reply]
Thanks @Albrecht-S 

I very much prefer github discussions over google groups :)
